GLMakie/src/drawing_primitives.jl

```diff
@@ -260,16 +260,29 @@ function draw_atomic(screen::Screen, scene::Scene, @nospecialize(x::Union{Scatte
     end
 end
 
+
+_mean(xs) = sum(xs) / length(xs) # skip Statistics import
+
+gl_linecap(::Nothing) = 0
+function gl_linecap(cap::Symbol)
+    cap == :square   && return 5
+    cap == :round    && return 4
+    # default to no cap
+    return 0
+end
+
 function draw_atomic(screen::Screen, scene::Scene, @nospecialize(x::Lines))
     return cached_robj!(screen, scene, x) do gl_attributes
         linestyle = pop!(gl_attributes, :linestyle)
+        linecap = pop!(gl_attributes, :linecap)
         data = Dict{Symbol, Any}(gl_attributes)
+        data[:linecap] = map(gl_linecap, linecap)
```

Comment:
I'm doing the conversion here wrong. The linecap input Observable is strongly typed at this point (I assume from `lift_convert`) so when you try changing the value from `nothing` to `:round` it errors. How do I do this correctly?

Reply:
You should turn that into a `convert_attributes` overload, so that lift_convert converts it to a stable type...So something like:
```julia
function convert_attribute(cap::Union{Nothing, Symbol}, ::key"linecap")
    cap == :square   && return 5
    cap == :round    && return 4
# TODO maybe error on wrong symbols?
    # default to no cap (nothing included)
    return 0
end
```

Reply:
Would this also trigger for CairoMakie?

Reply:
Yes. I did a `nothing -> :butt` conversion instead so CairoMakie can have more readable Symbols.

Reply:
CairoMakie doesn't need to apply the convert ;)

Reply:
Adding this results in getting numbers in CairoMakie though, which I'd rather not...